custom_components/zaptec/misc.py

```diff
@@ -91,6 +92,48 @@ def frame_decoder():
             raise AttributeError(f"Unknown record type {hex(record_type)}")
 
 
+# Format: <Year>-<Month>-<Day>T<Hours>:<Minutes>:<Seconds>,<Milliseconds><Time Zone>
+RE_OCMF_TIMESTAMP = re.compile(
+    r"^(\d{4})-(\d{2})-(\d{2})T(\d{2}):(\d{2}):(\d{2}),(\d{3})([+\-]\d{2}):(\d{2}) .*$"
+)
+
+
+def get_ocmf_latest_reader_value(data: dict) -> int:
+    """Return the latest reader value from OCMF data."""
+
+    if not isinstance(data, dict):
+        return 0
+    if "RD" not in data:
+        return 0
+
+    # Find the latest reading
+    readings = []
+    for reading in data["RD"]:
+        value = reading.get("RV")
+        if value is None:
+            continue
+        ts = None
+        m = RE_OCMF_TIMESTAMP.match(reading.get("TM", ''))
+        if m:
+            ts = datetime(
+                int(m[1]),  # Year
+                int(m[2]),  # Month
+                int(m[3]),  # Day
+                int(m[4]),  # Hours
+                int(m[5]),  # Minutes
+                int(m[6]),  # Seconds
+                int(m[7]) * 1000,  # Milliseconds
+                tzinfo=timezone(timedelta(hours=int(m[8]), minutes=int(m[9]))),  # Time zone
+            )
+            readings.append((ts, value))
+
+    # Sort readings by timestamp
+    readings.sort(key=lambda x: x[0])
+    if readings:
+        return readings[-1][1]  # Return the latest value
+    return 0
```

Comment:
Aren't we overcomplicating this? This is an increasing value that will never decrease, isn't it? So we just need the maximum value, right?